task/ecosystem-cert-preflight-checks/0.2/ecosystem-cert-preflight-checks.yaml

```diff
@@ -89,6 +90,25 @@ spec:
         printf "%s" "${_ARTIFACT_TYPE}" > "$(step.results.artifact-type.path)"
         echo "Introspection concludes that this artifact is of type \"${_ARTIFACT_TYPE}\"."
 
+    - name: generate-container-auth
+      image: quay.io/konflux-ci/oras:latest@sha256:fdf6e7156d4dec42ab2d20cf229eaf4bef886e93b1fd562d7734dc6a5bddd6e8
+      env:
+      - name: PARAM_IMAGE_URL
+        value: $(params.image-url)
+      results:
+        - name: auth-json-path
+          description: Path to auth.json
+      script: |
+        _AUTH_JSON_PATH="/auth/auth.json"
+        echo "Selecting auth for $PARAM_IMAGE_URL"
+        select-oci-auth "$PARAM_IMAGE_URL" > "${_AUTH_JSON_PATH}"
```

Comment:
In _this_ image, this script assumes the AUTHFILE to use is at ~/.docker/config.json - can we document that here, for the inevitability that this will change?

I don't think we have any expectations that this script's behavior won't change, given we don't create this image. I think it makes sense to document at least the expectation that the parse configuration file is the one I listed above.

Reply:
ref: https://github.com/konflux-ci/oras-container/blob/main/hack/select-oci-auth.sh

Reply:
```suggestion
        # `select-oci-auth` here assumes the input credentials are at path ~/.docker/config.json
        select-oci-auth "$PARAM_IMAGE_URL" > "${_AUTH_JSON_PATH}"
```